featuretools/tests/computational_backend/test_calculate_feature_matrix.py

```diff
@@ -1535,9 +1521,10 @@ def test_string_time_values_in_cutoff_time(es):
 
 
 # TODO: Dask version fails (feature matrix is empty)
+# TODO: Koalas version fails (koalas groupby agg doesn't support custom functions)
 def test_no_data_for_cutoff_time(mock_customer):
-    if any(isinstance(entity.df, dd.DataFrame) for entity in mock_customer.entities):
-        pytest.xfail('Dask version fails, returned feature matrix is empty')
+    if any(isinstance(entity.df, (dd.DataFrame, ks.DataFrame)) for entity in mock_customer.entities):
+        pytest.xfail("Dask fails because returned feature matrix is empty; Koalas doesn't support custom agg functions")
```

Comment:
well.. actually Koalas support custom aggregate functions through PySpark - it's way more flexible than Dask in this regard 
https://spark.apache.org/docs/3.0.0/sql-ref-functions-udf-aggregate.html

Reply:
I think we can use `.groupby(...).apply(...)`, see https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.groupby.GroupBy.apply.html#databricks.koalas.groupby.GroupBy.apply